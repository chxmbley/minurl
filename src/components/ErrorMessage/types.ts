/** Props for ErrorMessage component */
export type ErrorMessageProps = {
  /** CSS class to apply to the component */
  className?: string;
  /** Title of the error message */
  title?: string;
};
