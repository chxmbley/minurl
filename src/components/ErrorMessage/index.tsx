import cn from 'classnames';
import { ExclamationCircleIcon } from '@heroicons/react/solid';
import type { FC } from 'react';
import type { ErrorMessageProps } from '~components/ErrorMessage/types';

/** Container to semantically display an error message */
const ErrorMessage: FC<ErrorMessageProps> = ({ children, className, title = 'oops' }) => (
  <div
    className={cn(
      className,
      'flex border border-red-300 bg-red-800 bg-opacity-10 text-red-800 dark:text-red-50 p-4 rounded',
    )}>
    <ExclamationCircleIcon className="w-5 h-5 mr-2 mt-1 text-red-600 dark:text-red-400" />
    <div className="text-sm">
      <h3 className="block text-lg font-bold text-red-600 dark:text-white mb-1">{title}</h3>
      {children}
    </div>
  </div>
);

export default ErrorMessage;
