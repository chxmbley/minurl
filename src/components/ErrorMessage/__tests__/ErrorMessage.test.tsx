import { render } from '@testing-library/react';
import ErrorMessage from '~components/ErrorMessage';

test('renders an ErrorMessage component', () => {
  const { container } = render(<ErrorMessage />);
  expect(container).toMatchSnapshot();
});

test('adds a provided className to the component', () => {
  const customClassName = 'my-custom-classname';
  const { container } = render(<ErrorMessage className={customClassName} />);
  expect(container).toMatchSnapshot();
});

test('renders text content children', () => {
  const { container } = render(<ErrorMessage>Hello world</ErrorMessage>);
  expect(container).toMatchSnapshot();
});

test('renders React component children', () => {
  const { container } = render(
    <ErrorMessage>
      <strong>Hello world</strong>
    </ErrorMessage>,
  );

  expect(container).toMatchSnapshot();
});

test('overrides the default title when a title prop is provided', () => {
  const { container } = render(<ErrorMessage title="Test" />);
  expect(container).toMatchSnapshot();
});
