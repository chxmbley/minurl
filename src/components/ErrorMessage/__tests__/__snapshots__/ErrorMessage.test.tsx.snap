// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds a provided className to the component 1`] = `
<div>
  <div
    class="my-custom-classname flex border border-red-300 bg-red-800 bg-opacity-10 text-red-800 dark:text-red-50 p-4 rounded"
  >
    <svg
      aria-hidden="true"
      class="w-5 h-5 mr-2 mt-1 text-red-600 dark:text-red-400"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7 4a1 1 0 11-2 0 1 1 0 012 0zm-1-9a1 1 0 00-1 1v4a1 1 0 102 0V6a1 1 0 00-1-1z"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="text-sm"
    >
      <h3
        class="block text-lg font-bold text-red-600 dark:text-white mb-1"
      >
        oops
      </h3>
    </div>
  </div>
</div>
`;

exports[`overrides the default title when a title prop is provided 1`] = `
<div>
  <div
    class="flex border border-red-300 bg-red-800 bg-opacity-10 text-red-800 dark:text-red-50 p-4 rounded"
  >
    <svg
      aria-hidden="true"
      class="w-5 h-5 mr-2 mt-1 text-red-600 dark:text-red-400"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7 4a1 1 0 11-2 0 1 1 0 012 0zm-1-9a1 1 0 00-1 1v4a1 1 0 102 0V6a1 1 0 00-1-1z"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="text-sm"
    >
      <h3
        class="block text-lg font-bold text-red-600 dark:text-white mb-1"
      >
        Test
      </h3>
    </div>
  </div>
</div>
`;

exports[`renders React component children 1`] = `
<div>
  <div
    class="flex border border-red-300 bg-red-800 bg-opacity-10 text-red-800 dark:text-red-50 p-4 rounded"
  >
    <svg
      aria-hidden="true"
      class="w-5 h-5 mr-2 mt-1 text-red-600 dark:text-red-400"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7 4a1 1 0 11-2 0 1 1 0 012 0zm-1-9a1 1 0 00-1 1v4a1 1 0 102 0V6a1 1 0 00-1-1z"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="text-sm"
    >
      <h3
        class="block text-lg font-bold text-red-600 dark:text-white mb-1"
      >
        oops
      </h3>
      <strong>
        Hello world
      </strong>
    </div>
  </div>
</div>
`;

exports[`renders an ErrorMessage component 1`] = `
<div>
  <div
    class="flex border border-red-300 bg-red-800 bg-opacity-10 text-red-800 dark:text-red-50 p-4 rounded"
  >
    <svg
      aria-hidden="true"
      class="w-5 h-5 mr-2 mt-1 text-red-600 dark:text-red-400"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7 4a1 1 0 11-2 0 1 1 0 012 0zm-1-9a1 1 0 00-1 1v4a1 1 0 102 0V6a1 1 0 00-1-1z"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="text-sm"
    >
      <h3
        class="block text-lg font-bold text-red-600 dark:text-white mb-1"
      >
        oops
      </h3>
    </div>
  </div>
</div>
`;

exports[`renders text content children 1`] = `
<div>
  <div
    class="flex border border-red-300 bg-red-800 bg-opacity-10 text-red-800 dark:text-red-50 p-4 rounded"
  >
    <svg
      aria-hidden="true"
      class="w-5 h-5 mr-2 mt-1 text-red-600 dark:text-red-400"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7 4a1 1 0 11-2 0 1 1 0 012 0zm-1-9a1 1 0 00-1 1v4a1 1 0 102 0V6a1 1 0 00-1-1z"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="text-sm"
    >
      <h3
        class="block text-lg font-bold text-red-600 dark:text-white mb-1"
      >
        oops
      </h3>
      Hello world
    </div>
  </div>
</div>
`;
